src/actions/actions.ts

```diff
@@ -3615,7 +3615,7 @@ abstract class MoveByScreenLine extends BaseMovement {
       let stop = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.end);
 
       // We want to swap the cursor start stop positions based on which direction we are moving, up or down
-      if (start.line < position.line) {
+      if (start.line < position.line && this.movementType === "up") {
```

Comment:
I will take a peek to have a better answer for you, I was just playing wack a mole.